l10n_ch_import_pain000/models/undo_payment_line.py

```diff
@@ -1,61 +1,119 @@
+        """
+        This method is used to post message on the invoices that have been
+        deleted from the payment order and it post a message too on the
+        payment order for each deleted invoice.
+
+        :param data_supp: additional information used in the message
+        :param account_move_lines: used to find the invoices where to post
+        the message
+        :param payment_order_obj: used to post the message for each
+        deleted invoices
+        :return: -
+        """
+        if data_supp and 'add_tl_inf' in data_supp:
+
+            for account_move_line in account_move_lines:
+                # Create a link to the invoice that was removed
+                url = '<a href="web#id={}&view_type=form&model=' \
+                      'account.invoice&action=241">{}</a>'. \
+                    format(account_move_line.invoice_id.id,
+                           account_move_line.invoice_id.move_name)
+                # Add a message to the invoice
+                account_move_line.invoice_id.message_post(
+                    "The invoice has been removed from the payment "
+                    "order because: " +
+                    data_supp['add_tl_inf'])
+                # Add a message to the payment order
+                payment_order_obj.message_post(
+                    url + " has been removed because : " + data_supp[
+                        'add_tl_inf'])
+
+        else:
+            for account_move_line in account_move_lines:
+                # Create a link to the invoice that was removed
+                url = '<a href="web#id={}&view_type=form&model=account. ' \
+                      'invoice&action=241">{}</a>'. \
+                    format(account_move_line.invoice_id.id,
+                           account_move_line.invoice_id.move_name)
+                # Add a message to the invoice
+                account_move_line.invoice_id.message_post(
+                    "The invoice has been removed from the payment order,"
+                    " no reason given.")
+                # Add a message to the payment order
+                payment_order_obj.message_post(
+                    url + " has been removed no reason given.")
+
+    def _process_invoice_remove(self, account_payment_lines, payment_order_obj,
+                                data_supp):
+        """
+        This method unreconcile the given move lines and remove the payment.
+
+        :param account_payment_lines: The payment to remove
+        :param payment_order_obj: The payment order where to remove the payment
+        :param data_supp: additional information used in the message
+        :return: -
+        """
+
+        # Retrieve all account_move_line(s) associated with the previous
+        # account_payment_line(s)
+        account_move_lines = account_payment_lines.mapped(
+            'move_line_id')
+
+        full_reconcile_id = account_move_lines.mapped(
+            'full_reconcile_id').id
```

Comment:
Are you sure there will always be only one reconcile object ? Because your code will break in the case where there are mutliple reconcile records.